### DISCUSSION TITLE
HTTP/2 on Mac and HTTP/2 WebSockets

### DISCUSSION THREAD

**Discussion by alexandrehtrb** [dotnet/runtime#75096] [Q&A] [answered]
Hello,

I would like to know:

1) Is there any planning on HTTP/2 support for MacOS?
2) Will .NET 7 support HTTP/2 WebSockets? If yes, in which platforms?

**Comment by danmoseley** [comment]
@karelz

**Comment by karelz** [comment, marked as answer]
1. Client-side HTTP/2 works just fine on MacOS. Server-side HTTP/2 does not work on MacOS (see #27727) as there are structural problems in the MacOS available APIs and what .NET relies on.
2. .NET 7.0 RC1 will support HTTP/2 WebSockets -- see https://github.com/dotnet/core/issues/7716#issuecomment-1227162510
